### ISSUE TITLE
Remove support for versions below draft 11

### ISSUE THREAD

**Issue by TimoGlastra** [Sphereon-Opensource/OID4VCI#75]
When developing on this library, over time it has lost some of the simplicity due to several formats and protocol versions being supported.

We have targeted Draft 8 before, but now are mainly working on Draft 11 and possibly newer drafts (maybe 19 or another one). To improve maintainability, I propose support for Draft 8 and 9 is removed. This will allow us to remove a lot of transformation code, and also simplify a lot of small issues currently encountered with this library.

Thoughts on removing support for drafts below 11? People still wanting to support these older drafts could stick to an older version of the library for now

**Comment by nklomp** [comment]
Yeah agreed. For VP/SIOP we have to be a bit more careful, but for VCI it is not problem. We do have version 8 or 9 in the DIIP profile though 😱

**Comment by TimoGlastra** [comment]
Hmm good point. But I thought we used draft 11 for DIIP. We showed interop for DIIP based on draft 11 (as that's the only one your issuer supports)

**Comment by nklomp** [comment]
Yeah, somehow we managed to not specify the actual version for VCI in DIIP 😂 
Given the issuer has always been draft 11 anyway, as it was developer later then the client lib, let's just drop v8/v9 support altogether.

I think one discussion we need to have is whether we will wait for the next version in which the inline offers are removed and replaced by id's only or not and then only support that version

**Comment by TimoGlastra** [comment]
It seems draft 12 has just been published: https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html

Lots of changes here, with some notable ones the offer `credentials` being only strings now, and the credentials_supported being a map again, so we're back to what it was a couple of drafts ago it seems.

Do we just want to support that version? There's a lot of interop demonstrations currently based on draft 11, so I think dropping draft 11 will have a bigger impact. But with the vast amount of changes, it would be nice to just get rid of everything but the latest draft.

**Comment by nklomp** [comment]
BTW small update on this one 

After the current SD-JWT work has landed, we will be updating this monorepo and also bring in the SIOP /OID4VP as modules. Reason is there is certain overlap and then having the modules next to one-another helps. In certain scenarios for instance doing a SIOP call from within a VCI context makes sense. That is of course next to having shared interfaces between the two of them.

We will also drop support for non-latest versions when starting that work as the specs seem to become a bit more stable. This means we will remove most of the version discovery, which should simplify the code drastically.
Given work on MDL/Mdoc will be landing as well, we want to simplify the burden of supporting a lot of versions. It has served us well in interop cases up till this point, but there simply are too many draft versions, having a significant impact on the code and potential for bugs etc.
